### ISSUE TITLE
Intergrate with Webform

### ISSUE THREAD

**Issue by ghost** [backdrop-contrib/addressfield#23]
It'd be nice if this module integrated with [Webform](https://github.com/backdrop-contrib/webform) to allow address fields there. Perhaps the Webform integration code from https://www.drupal.org/project/addressfield_tokens could be used...?

**Comment by jenlampton** [comment]
Are the tokens the only thing that is needed? Or do we need a webform "address" component?

**Comment by ghost** [comment]
I can't remember what I was wanting to use this for now, but I'm guessing it was needing to add an address field in a webform...

**Comment by alanmels** [comment]
Relevant discussion that references _[addressfield_tokens](https://www.drupal.org/project/addressfield_tokens)_ : https://www.drupal.org/project/addressfield/issues/1252974

**Comment by john-haus** [comment]
not sure if this is dead, but it would be a great addition to the webform module.

**Comment by argiepiano** [comment]
I think it'd be great to try to integrate some of the functionality provided by `addressfield_token`. However, `addressfield_token` depends on `entity_token` which depends on D7 Entity API (Entity Plus in Backdrop).  Adding this dependency is a big change to Addressfield. 

Perhaps someone can give it a try to add at least the webform component without Entity Plus dependency? Otherwise I'd say a separate contrib module would be the way to go.

**Comment by argiepiano** [comment]
I have created a PR that provides a Webform component, borrowed from D7 Addressfield Tokens. Please note: I have ported only the webform component - not the rest of the functionality of that Drupal module. I have also modified the README to give credit to that module.

PR #42.

@padprint, @BWPanda, @jenlampton, I'd be grateful if you could test the PR.

**Comment by jenlampton** [comment]
I agree with supporting the webform component, but I would also prefer to avoid the chain of dependencies. I think addressfield_token is probably best kept separate. entity_token has performance implications for sites with lots of entities/fields and if that were a dependency it could prevent people from being able to use this module.  (if, however, we could find a way to make it optional, that would be another story...)

I've tested the webform component and it works really well :) I left some minor feedback on the PR.

**Comment by argiepiano** [comment]
Thanks, @jenlampton. I've made the changes you requested! After you test, we can go ahead and merge.

**Comment by argiepiano** [comment]
Hi @jenlampton. I'd like to merge this PR and do a new release of Addressfield. Are you OK with that?

**Comment by alanmels** [comment]
Also see: https://github.com/backdrop-contrib/addressfield/issues/35

**Comment by yorkshire-pudding** [comment]
Hi @argiepiano and @jenlampton - I've tested this PR on a site I'm building and it mostly works well.

If I make address field required, select a single country in Available countries, also select as default (or don't select - makes no difference) and check "Hide country field (when only one is available)" 
![image](https://github.com/backdrop-contrib/addressfield/assets/20357797/c7df167b-657f-4042-92c7-20ccc462eda2)

then Country still shows
![image](https://github.com/backdrop-contrib/addressfield/assets/20357797/50890853-6afa-414a-b2a0-78b21cf6d7f7)

I confirmed that this behaviour works as expected on the node form.

I've added a [suggestion to the PR](https://github.com/backdrop-contrib/addressfield/pull/42/files#r1412379293) that fixes this

![image](https://github.com/backdrop-contrib/addressfield/assets/20357797/78170df4-5c8c-4239-8259-aa1e5301cfbd)

**Comment by argiepiano** [comment]
@yorkshire-pudding, thanks for taking the time to test this. Do you think this is RTBC after your testing? (I'm not currently using this module).

**Comment by yorkshire-pudding** [comment]
@argiepiano - yes. I've given it another test run to be certain, but yes, I think this is ready and would be a very welcome addition to the module and to the webform functionality.
Thanks

**Comment by argiepiano** [comment]
Thanks! If there are no concerns from @jenlampton I'll merge over the weekend.

**Comment by yorkshire-pudding** [comment]
Bump @argiepiano

**Comment by argiepiano** [comment]
Bump @jenlampton! As co-maintainer, I'd like to get @jenlampton's OK.

**Comment by argiepiano** [comment]
@jenlampton, perhaps you are not keen on providing this webform element? If so, let me know and I'll create a separate contrib module for this.

**Comment by argiepiano** [comment]
I have created a contrib module that provides this functionality: 
https://github.com/backdrop-contrib/addressfield_wf

Closing here.
